tripy/examples/nanogpt/weight_loader.py

```diff
@@ -50,7 +49,7 @@ def load_weights_from_hf(model, model_type, dtype):
                 weight = hf_state_dict[key].t().contiguous()
         if "ln" not in key:
             weight = weight.to(torch_dtype)
-        param = tp.Parameter(weight)
+        param = tp.Parameter(tp.Tensor(weight))
```

Comment:
We might want to make an exception for `tp.Parameter`, since it might be convenient. Perhaps we could have a switch in the decorator for permitting external input types (false by default)?

Reply:
Yeah, a switch makes sense.

Reply:
Actually, instead of a switch, can we just make the `Parameter` constructor create the tensor if the input is not a tensor? Otherwise we're over-complicating the decorator.

Reply:
I went with the latter approach.